components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java

```diff
@@ -714,97 +722,104 @@ private boolean retrieveFileToStreamInBody(String name, Exchange exchange) throw
     @SuppressWarnings("unchecked")
     private boolean retrieveFileToFileInLocalWorkDirectory(String name, Exchange exchange) throws GenericFileOperationFailedException {
         File temp;
-        File local = new File(endpoint.getLocalWorkDirectory());
+        boolean skipRetrieval = false;
         OutputStream os;
         GenericFile<ChannelSftp.LsEntry> file =
                 (GenericFile<ChannelSftp.LsEntry>) exchange.getProperty(FileComponent.FILE_EXCHANGE_FILE);
         ObjectHelper.notNull(file, "Exchange should have the " + FileComponent.FILE_EXCHANGE_FILE + " set");
-        try {
-            // use relative filename in local work directory
-            String relativeName = file.getRelativeFilePath();
-
-            temp = new File(local, relativeName + ".inprogress");
-            local = new File(local, relativeName);
-
-            // create directory to local work file
-            local.mkdirs();
-
-            // delete any existing files
-            if (temp.exists()) {
-                if (!FileUtil.deleteFile(temp)) {
-                    throw new GenericFileOperationFailedException("Cannot delete existing local work file: " + temp);
+        if (file.getFile() != null && file.getFile() != null
```

Comment:
Is `file.getFile() != null` duplicated intently? And `file.getFile().getAttrs()` is not null-checked before you invoke a method call on it.